### PR TITLE
Tweak expected message to explain what it's actually signifying

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @estebank since you added this

### CODE REVIEW

File: tests/ui/associated-types/defaults-specialization.stderr [hunk 1]
```diff
@@ -29,7 +29,7 @@ error[E0053]: method `make` has an incompatible type for trait
   --> $DIR/defaults-specialization.rs:35:18
    |
 LL |     default type Ty = bool;
-   |     ----------------------- expected this associated type
```
Comment (compiler-errors) [inline tests/ui/associated-types/defaults-specialization.stderr:32]: Side-note: The whole point of this E0053 is that the type `bool` is *not* expected.

**Comment by ehuss** [comment]
This PR has been re-added to the queue.

**Pull request merged**